src/webloader.js

```diff
@@ -0,0 +1,126 @@
+import { generate } from './guids.js';
+import EvolvAssetManager from './index.js';
+import EvolvClient from "@evolv/javascript-sdk";
+
+
+function ensureId(evolv, key, session) {
+	let id;
+	if (evolv.context && key in evolv.context) {
+		id = evolv[key];
+	} else {
+		id = evolv.retrieve(key, session);
+		if (!id) {
+			id = generate();
+			evolv.store(key, id, session);
+		}
+	}
+	return id;
+}
+
+function currentScript() {
+	if (document.currentScript) {
+		return document.currentScript;
+	}
+
+	for (let i = 0; i < document.scripts.length; i++) {
+		const script = document.scripts[i];
+		if (script && script.dataset && 'env' in script.dataset) {
+			return script;
+		}
+	}
```

Comment:
It's possible that there is no script tag with a `data-evolv-environment` attribute. Therefore throw the error that you throw later on line 74 here. I also modified the log message slightly to have consistency with the execution plan, which uses a `[Evolv]` prefix on log messages.

```suggestion
		[Evolv] Environment not specified
	}
```